### PR TITLE
SC-1886: Enhance tenant preferences read and update api

### PR DESCRIPTION
**Pull request by aishwa8141**
End Point : **/v2/org/preferences/update**
**Request** : `{
    "request":{
        "orgId":"",
        "key": "",
        "data": {
        }
    }
}`

**Response**: `{
    "id": "api.org.preferences.update",
    "ver": "v1",
    "ts": "2020-08-06 23:24:56:485+0530",
    "params": {
        "resmsgid": null,
        "msgid": "a670b165-d8eb-4b02-b98a-ee47d7e34f5f",
        "err": null,
        "status": "success",
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "response": "SUCCESS"
    }
}`

End point: **/v2/org/preferences/read**

**Request**: `{
    "request":{
        "orgId":"",
        "key": ""
    }
}`

**Response**: `{
    "id": "api.org.preferences.read",
    "ver": "v1",
    "ts": "2020-08-06 23:25:12:753+0530",
    "params": {
        "resmsgid": null,
        "msgid": "a2a5d2fb-7ce1-4280-9bf1-d12d289a7822",
        "err": null,
        "status": "success",
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "response": {
            "updatedBy": "2b5766bc-b6d0-4f8d-a677-818e006146eb",
            "data": {
            },
            "createdBy": "2b5766bc-b6d0-4f8d-a677-818e006146eb",
            "updatedOn": 1596736496420,
            "createdOn": 1596736476856,
            "key": "",
            "orgId": ""
        }
    }
}`

### CODE REVIEW

File: actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/learner/actors/tenantpreference/TenantPreferenceManagementActor.java [hunk 1]
```diff
@@ -64,86 +64,94 @@ public void onReceive(Request request) throws Throwable {
...
-          preferenceObj.put(JsonKey.KEY, key);
-          preferenceObj.put(JsonKey.DATA, data);
-          cassandraOperation.updateRecord(
-              tenantPreferenceDbInfo.getKeySpace(),
-              tenantPreferenceDbInfo.getTableName(),
-              preferenceObj);
-          responseList.add(getResponseMap(orgId, key, null));
-        }
+    List<Map<String, Object>> preferencesList = getTenantPreferencesFromDB(orgId, key);
+    if (CollectionUtils.isEmpty(preferencesList)) {
```
Comment (indrajra) [inline actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/learner/actors/tenantpreference/TenantPreferenceManagementActor.java:124]: Recall we may like to reset to empty - to indicate no preferences. Should this exception be needed. I think we can allow empty map.
Reading this am thinking we must have a delete API as well - just to remove this in total.

Comment (aishwa8141) [reply actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/learner/actors/tenantpreference/TenantPreferenceManagementActor.java:124]: @indrajra this preference we are reading from DB, based on key and orgId in the update req, to update the data the preference should be present in DB, if not we have to throw not found exception right?

Comment (aishwa8141) [reply actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/learner/actors/tenantpreference/TenantPreferenceManagementActor.java:124]: if they need to update empty map data, then I should not check emptiness, that I am doing in TenantPreferenceValidator

Comment (indrajra) [reply actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/learner/actors/tenantpreference/TenantPreferenceManagementActor.java:124]: ok.

**Pull request merged**